### PR TITLE
Support disambiguation of targets with names that only differ by case

### PR DESCRIPTION
**Pull request by cgrindel**

### CODE REVIEW

File: tools/generator/test/DisambiguateTargetsTests.swift [hunk 1]
```diff
@@ -23,12 +23,23 @@ final class DisambiguateTargetsTests: XCTestCase {
...
             ),
+            // The following targets only differ by case
+            "C 1": Target.mock(
+                label: "//c:A",
+                product: .init(type: .staticLibrary, name: "A", path: "")
+            ),
+            "C 2": Target.mock(
+                label: "//c:a",
+                product: .init(type: .application, name: "a", path: "")
+            ),
```
Comment (brentleyjones) [inline tools/generator/test/DisambiguateTargetsTests.swift:34]: Can you add an example to `test_productType` to show that it does this for non-label based results as well?

File: tools/generator/test/DisambiguateTargetsTests.swift [hunk 2]
```diff
@@ -60,11 +71,20 @@ final class DisambiguateTargetsTests: XCTestCase {
...
+            "C 2": Target.mock(
+                product: .init(type: .application, name: "c", path: "")
+            ),
         ]
         let expectedTargetNames: [TargetID: String] = [
             "A 1": "A (Library)",
             "A 2": "A (App)",
             "B": "B",
+            "C 1": "C (Library)",
+            "C 2": "c (App)",
```
Comment (cgrindel) [inline tools/generator/test/DisambiguateTargetsTests.swift:87]: @brentleyjones I just want to confirm that this is the output you would expect for the `producType` test. If so, I will need to modify the following conditional 
```swift
if labelsByName[normalizedName]!.count == 1 {
```

Comment (brentleyjones) [reply tools/generator/test/DisambiguateTargetsTests.swift:87]: That is the expected output, yes.

Comment (brentleyjones) [reply tools/generator/test/DisambiguateTargetsTests.swift:87]: That condition is fine. We just need to store normalized labels in it here:
```
labelsByName[target.name, default: []].insert(target.label)
```

Comment (brentleyjones) [reply tools/generator/test/DisambiguateTargetsTests.swift:87]: And then the label test will need to use unique (not just case differences) labels.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**